src/OpenTelemetry.Abstractions/Trace/TracerExtensions.cs

```diff
@@ -29,7 +30,7 @@ public static class TracerExtensions
         /// <returns>Span instance.</returns>
         public static ISpan StartRootSpan(this ITracer tracer, string operationName)
         {
-            return tracer.StartRootSpan(operationName, SpanKind.Internal, PreciseTimestamp.GetUtcNow(), null);
+            return tracer.StartRootSpan(operationName, SpanKind.Internal, PreciseTimestamp.GetUtcNow(), null as Func<IEnumerable<Link>>);
```

Comment:
bummer. we have an ambiguous API that requires to specify type for null (lambda or links). I think let's keep it  for now, but do better with https://github.com/open-telemetry/opentelemetry-dotnet/issues/297

Reply:
#297 make sense to me. Should we abandon this PR then?
We're doing this to avoid the allocation for IEnumerable in the NoOp scenario. It's quite an API addition for a optimization used when using only the API and not the SDK (I assume not the most  common scenario).

Reply:
we'll have another PR for #297, keep this one please!

Reply:
OK. Thanks